### PR TITLE
Fix PR test library import

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@karajas Check it out! Tests!  Also PTAL :)

**Comment by karajas** [comment]
Awesome! Where is intended to be run? Like would I be able to test this from outside dotnet-ci to see if I've broken any basic functionality?

**Comment by mmitche** [comment]
@karajas Nope.  This requires the CI server to test.  You also need to push to this repo for the PR so that the trusted libraries will load.  But at least it's a pre-merge check.

File: tests/pipeline/pipeline-tests.groovy [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+// To figure out what we import, we need to know whether this is a PR or not.
+// If a PR, and GitHub, we grab the source branch, which must live in the same repo.
+// Note that we can't use the library isPR for this.
+boolean isPRTest = false
+String repository = env["ghprbGhRepository"]
+String libraryImportBranch
+if (repository != null && repository != "") {
+    echo "This is a GitHub PR test"
+    // Check that the PR source branch came from the correct location (otherwise the tests are expected to fail)
+    if (repository != 'dotnet/dotnet-ci') {
```
Comment (karajas) [inline tests/pipeline/pipeline-tests.groovy]: This is when we invoke this pipeline on non dotnet-ci runs?

Comment (mmitche) [inline tests/pipeline/pipeline-tests.groovy]: The pipeline itself is pulling the library dependency from the PR source branch.  A few lines below you can see that it loads the pipeline.

**Pull request merged**